src/viking.js

```diff
@@ -1,11 +1,92 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+
+  receiveDamage(damage) {
+    this.health -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`;
+    } else if (this.health <= 0) {
+      return `${this.name} has died in act of combat`;
+    }
+  }
+
+  battleCry() {
+    return "Odin Owns You All!";
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  constructor(health, strength) {
+    super(health, strength);
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`;
+    } else if (this.health <= 0) {
+      return `A Saxon has died in combat`;
+    }
+  }
+}
 
-// War
-class War {}
+// // War
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+  addViking(Viking) {
+    this.vikingArmy.push(Viking);
+  }
+  addSaxon(Saxon) {
+    this.saxonArmy.push(Saxon);
+  }
+  vikingAttack() {
+    var saxonNewHealth = this.saxonArmy[0].receiveDamage(
+      this.vikingArmy[0].strength
+    );
+    if (this.saxonArmy[0].health <= 0) {
+      this.saxonArmy.length -= 1;
+    }
+    return saxonNewHealth;
+  }
```

Comment:
Works well, but in that case it's always the first viking of the vikingArmy array attacking the first saxon of the saxonArmy array 😉 And then you remove the last saxon from the saxonArray if the first one is dead.
To be more precise, you could try choosing a saxon and viking at random like specified in the lab instructions, and removing that exact random saxon in case he's dead (although the jasmine tests were not checking it)